Firmware/printer.cfg

```diff
@@ -0,0 +1,275 @@
+# This file contains common pin mappings for the BIGTREETECH SKR
+# MINI 2.0. To use this config, the firmware should be compiled for 
+# the STM32F103 with a "28KiB bootloader". Also select "enable extra 
+# low-level configuration options" and configure "GPIO pins to set 
+# at micro-controller startup" to "!PA14".
+
+# The "make flash" command does not work on the SKR mini. Instead,
+# after running "make", copy the generated "out/klipper.bin" file to a
+# file named "firmware.bin" on an SD card and then restart the SKR
+# mini with that SD card.
+
+# See the example.cfg file for a description of available parameters.
+
+
+
+# Required include for Mainsail
+[include mainsail.cfg]
+
+
+[mcu]
+##	Obtain mcu value by "ls -l /dev/serial/by-id/" 
+serial: /dev/serial/by-id/usb-Klipper_stm32f103xe_33FFDB055356343509851943-if00
+
+
+[stepper_x]
+step_pin: PB13
+dir_pin: PB12           # Add ! if moving opposite direction
+enable_pin: !PB14
+rotation_distance: 40
+microsteps: 16
+endstop_pin: PC0 
+position_endstop: 120
+position_max: 120
+homing_speed: 20        # Increase after initial setup, Max 100
+homing_retract_dist: 5
+homing_positive_dir: true
+
+[tmc2209 stepper_x]
+uart_pin: PC11
+tx_pin: PC10
+uart_address: 0
+interpolate: True
+run_current: .5         # For V0.1 spec NEMA14 40Ncm
+hold_current: .25
+sense_resistor: 0.110
+stealthchop_threshold: 500
+
+
+[stepper_y]
+step_pin: PB10
+dir_pin: PB2            # Add ! if moving opposite direction
+enable_pin: !PB11
+rotation_distance: 40
+microsteps: 16
+endstop_pin: PC1
+position_endstop: 120
+position_max: 120
+homing_speed: 20        # Increase after initial setup, Max 100
+homing_retract_dist: 5
+homing_positive_dir: true
+
+[tmc2209 stepper_y]
+uart_pin: PC11
+tx_pin: PC10
+uart_address: 2
+interpolate: True
+run_current: .5         # For V0.1 spec NEMA14 40Ncm
+hold_current: .25
+sense_resistor: 0.110
+stealthchop_threshold: 500
+
+
+[stepper_z]
+step_pin: PB0
+dir_pin: !PC5           # Remove ! if moving opposite direction
+enable_pin: !PB1
+rotation_distance: 8    # For T8x8 integrated lead screw
+microsteps: 16
+endstop_pin: PC2
+position_endstop: -.10
+position_max: 120
+position_min: -2
+homing_speed: 10    # Max 100
+second_homing_speed: 3.0
+homing_retract_dist: 3.0
+
+[tmc2209 stepper_z]
+uart_pin: PC11
+tx_pin: PC10
+uart_address: 1
+interpolate: True
+run_current: 0.37       # For V0.1 NEMA17 w/ integrated lead screw
+hold_current: 0.35
+sense_resistor: 0.110
+stealthchop_threshold: 500
+
+
+[extruder]
+step_pin: PB3
+dir_pin: PB4            # Add ! if moving opposite direction
+enable_pin: !PD2
+full_steps_per_rotation: 200 # 1.8deg motor
+rotation_distance: 22.67895  # For 5mm shaft driven Bondtech gearsets
+gear_ratio: 50:17       # For v0.1 Afterburner Mini
```

Comment:
Gear ratio is 50:10 for the mini AB with 10t pinion driving 50t BMG gear (this is needed or e-steps is wildly off).

Reply:
I started with 50:10 but the rotation_distance was 22.23. should I update both values?

Reply:
50:10 ratio and 22.67895 gets me within 1% of correct extrusion length without fiddling with e-steps, using the spec LDO motor.